### ISSUE TITLE
Layout & Tests

### ISSUE THREAD

**Issue by bramski** [alexk111/ph-challenge#3]
Your application layout is rather non-traditional.  So I am rather curious about your reasoning for this.

(1) You've organized by component.  Intermixing css, html, and JS in the same directories.
   - Most apps eventually need to minify and compress.  Could it lead to mistakes in this process?
   - Does it feel scalable?  Will it be manageable as the app grows?
 
(2) Tests are intermixed with src, this is something I've really infrequently seen.  you've made some effort to prevent them from getting in your build process.  Thus leading to more by convention than by organization testing process.  Which is interesting.  Many developers would find this foreign.  Thoughts?

In general it's very interesting for a project of this size it works, but what are your thoughts, why do it this way?  Is it better?  How else do you feel it is different?

**Comment by alexk111** [comment]
Yup, it's non-traditional. I came to that structure after working on a huge front-end app. Organizing app in the "classic" functional way (htmls, css, services, directives, controllers in separate dirs) had a bad readability and forced you to jump between lots of different dirs while working on a component, especially on a visual one consisting of at least four files (html, css, directive, tests). This produced a real headache while working on an app consisting of a hundred components. The proposed "component" structure is much better for huge apps, than the one offered by Angular team in their demo examples. 

> Most apps eventually need to minify and compress. Could it lead to mistakes in this process?

No. The structure I proposed will not lead to any mistakes in the process. There is a 'gulp build' command that already does the minification things. You may add a new component to the components dir and it will be compiled&minified without problems.

> Does it feel scalable? Will it be manageable as the app grows?

Yes, the initial reason of that structure is the ease of management for huge apps.

> Tests are intermixed with src

Yup. For TDD it is also convenient to have component's tests within that component's dir.

> Many developers would find this foreign.

I don't think it's a huge pay for a really better readability and scalability. Just group files by the app structure units instead of the file functions - it's not so hard :)
